db/view/view_update_generator.cc

```diff
@@ -33,28 +34,46 @@ future<> view_update_generator::start() {
             if (_sstables_with_tables.empty()) {
                 _pending_sstables.wait().get();
             }
-            while (!_sstables_with_tables.empty()) {
-                auto& [sst, t] = _sstables_with_tables.front();
+            for (auto& [t, sstables] : _sstables_with_tables) {
                 try {
                     schema_ptr s = t->schema();
-                    flat_mutation_reader staging_sstable_reader = sst->read_rows_flat(s, no_reader_permit());
-                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, _db, sst, _as), db::no_timeout);
+                    auto ssts = make_lw_shared(t->get_compaction_strategy().make_sstable_set(s));
+                    for (auto& sst : sstables) {
+                        ssts->insert(sst);
+                    }
+
+                    flat_mutation_reader staging_sstable_reader =
+                        ::make_range_sstable_reader(s,
+                                                    no_reader_permit(),
+                                                    std::move(ssts),
+                                                    query::full_partition_range,
+                                                    s->full_slice(),
+                                                    service::get_local_streaming_read_priority(),
+                                                    nullptr,
+                                                    ::streamed_mutation::forwarding::no,
+                                                    ::mutation_reader::forwarding::no);
+
+                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, *t, sstables, _as), db::no_timeout);
                     if (result == stop_iteration::yes) {
                         break;
                     }
                 } catch (...) {
+                    for (auto& sst : sstables) {
```

Comment:
Looks like this series needs a "fix indentation" commit. Also, GitHub presents indentation changes in a nice way, so there's no need for it really - it's fine to just use correct indentation from the start.

Reply:
It does, but I like to actually do those after an initial review round otherwise it's just wasted work.
I will include now